GitOps environment UI page validation

amitkrout opened the pull request:
Jira reference - https://issues.redhat.com/browse/ODC-5413

amitkrout commented:
@gajanan-more @sanketpathak Would please let me know what i am missing 
```
$ yarn run gherkin-lint
yarn run v1.22.10
$ /Users/amit/go/src/github.com/openshift/console/frontend/packages/gitops-plugin/integration-tests/node_modules/.bin/gherkin-lint
Could not find default config file ".gherkin-lintrc" in the working directory.
To use a custom name/path provide the config file using the "-c" arg.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

amitkrout commented on frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:
```diff
@@ -16,7 +19,10 @@ Feature: Applications managed by GitOps and Create secret
         @regression @manual
         Scenario: Create Secret: GO-01-TC02
-            Given user has imported gitops repo
+            # "user has imported gitops repo" looks too generic.
+            # There are several sub steps covered in - 
+            # https://github.com/redhat-developer/kam/tree/master/docs/journey/day1
+            Given user has imported gitops repo 
```
Do you want me to expand it ?
ping @gajanan-more @sanketpathak

amitkrout commented:
As per the screenshoot [1] expected in epic https://issues.redhat.com/browse/ODC-5157, I have added/updated the test scenario specific to environment page changes in developer console. However i do not see the same change on the latest (6th June) OCP 4.8 nightly build 

[1] 
<img width="495" alt="Screen Shot 2021-06-07 at 3 02 01 PM" src="https://user-images.githubusercontent.com/19851115/120993654-613d4700-c7a1-11eb-8ebd-14a06189770a.png">

@reginapizza Would you please have a look.

reginapizza commented:
@amitkrout you are seeing that view because the newest version of the gitops operator isn't be used I think? Only the newer version of the gitops operator will pass along the information for each environment's sync status and the lastest deployment time, otherwise it defaults to just a list of the dev environments and no time. Hope that clears up some confusion?

amitkrout commented:
> @amitkrout you are seeing that view because the newest version of the gitops operator isn't be used I think? Only the newer version of the gitops operator will pass along the information for each environment's sync status and the lastest deployment time, otherwise it defaults to just a list of the dev environments and no time. Hope that clears up some confusion?

@reginapizza Your guess is correct. I tested the secnarion on GitOps Version 1.1.1. I think the new console changes will be reflected in GitOps 1.1.2 patch release happening shortly.

@baijum We can merge this changes

baijum commented:
> @baijum We can merge this changes

@amitkrout ?!

makambalaji commented on frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:
```diff
@@ -2,18 +2,18 @@
+              And user has installed Pipelines operator
+              And user has installed Sealed Secrets operator in "cicd" namespace
+              And user has created Sealed Secrets instance of name "sealedsecretcontroller"
+              Then user is at developer perspective
```
```suggestion
              And user is at developer perspective
```
In Background section, Only Given statements should be added

amitkrout replied:
Done

makambalaji reviewed (changes requested):
One small modification needed

amitkrout commented:
> > @baijum We can merge this changes
> 
> @amitkrout ?!

Ahh... i am sorry @baijum. I suppose to add @makambalaji instead. 

@baijum Please ignore

amitkrout commented:
@makambalaji Addressed your review comment. Can you please check and merge if all look good.

amitkrout commented:
> @makambalaji Addressed your review comment. Can you please check and merge if all look good.

Also looking for developer (@reginapizza) review. @reginapizza Please apply ```/lgtm``` in the comment if the test scenario looks good.

reginapizza commented on frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:
```diff
@@ -22,17 +22,21 @@ Feature: Applications managed by GitOps and Create secret
               And user can see the Applications on the page
-             When user clicks on the application
-             Then user can see the application Details page
+             Then user can see the columns "Application Name", "Git repository", "Environments" and "Last deployment"
```
Based on their version of the GitOps operator, they will either see "Environments" (older; where all environments are in a comma-separated list) or "Environment Status" (newer; will have the # of environments for each status condition- Synced, OutOfSync, and Unknown)

Also, Application Name --> Application name (sentence case)

amitkrout replied:
@reginapizza Can you please share the support matrix. 

| GitOps Version  | OCP Version | Env Page  |
| ------------- | ------------- |  ------------- |
|   |   |   |
|   |   |   |

amitkrout replied:
If GitOps version has the dependency then i think it will be easy to handle in the test implementation code. May be i can rephrase the step as 

```Then user can see the columns attributes depending upon the "GioOps" version```

reginapizza replied:
working on confirming support matrix and I'll send that to you when I have it, but otherwise I think this statement is okay since it's generic.

reginapizza commented on frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:
```diff
@@ -22,17 +22,21 @@ Feature: Applications managed by GitOps and Create secret
+        Scenario: Application Details page for Applications: GO-01-TC04
+            Given user is on the Environments page
+              And user can see the Applications on the page
+             Then user can add new environments
```
User cannot add new environments directly through console, if that's what this is implying

amitkrout replied:
Yes, through its not possible. But using KAM user can add that. Let me rephrase it

amitkrout replied:
```
hen user can add new environments using kam cli
```
I think it makes more sense

reginapizza replied:
perfect, thank you!

amitkrout commented:
@reginapizza Thanks for the review :) 

I have address your review comments. Can you please go for another round of review.

reginapizza commented:
/lgtm

gajanan-more commented:
/lgtm

makambalaji commented:
/approve

reginapizza commented on frontend/packages/gitops-plugin/integration-tests/features/gitops/add-manifest-url.feature:
```diff
@@ -22,17 +22,21 @@ Feature: Applications managed by GitOps and Create secret
               And user can see the Applications on the page
-             When user clicks on the application
-             Then user can see the application Details page
+             Then user can see the columns attributes depending upon the GioOps "version"
```
sorry, just one thing, typo here on GitOps

amitkrout replied:
Done.
Thanks :)

amitkrout commented:
> /lgtm

@gajanan-more Please reapply it.

reginapizza commented:
/lgtm
